### PR TITLE
fix(runtime): Restore default signal handler after user handlers are unregistered

### LINKED ISSUE #7164: Signal default actions never restored after `sig.dispose()` called
Example:

```ts
const sig = Deno.signal(Deno.Signal.SIGINT);
// don't exit
const exitBlocker = setTimeout(() => {}, 100000);
await sig;
sig.dispose()
console.log("Signal disposed");
// Now if you try Ctrl-C it will still never exit the program, as default is never restored.
```

This is due to upstream Tokio behavior: See https://github.com/tokio-rs/tokio/blob/138eef352671aadffca304d62c69cb2582e64f2a/tokio/src/signal/unix.rs#L346-L361

It is debatable if we should reset to [`SIG_DFL`](https://man7.org/linux/man-pages/man2/sigaction.2.html) ourselves, but it will break Tokio's handling due to Tokio having no idea we secretly resets and it relies on the presence of some `SigInfo` struct to determine if we have ever registered for listeners, meaning future reregistration will no longer work.

While it is probably more of a Tokio issue here if we want the default handling back, it might be worthy for us to discuss even if reset to default is the right thing to do. Also we need tracking of this issue on our side.

Notice Node.js does seem to reset to default correctly:
```js
let counter = 0;
function handler() {
  console.log("Register");
  counter++;
  if (counter == 3) {
    console.log("Unregister");
    process.off('SIGINT', handler);
    // Ctrl-C after this kills the program
  }
}

process.on('SIGINT', handler);
setTimeout(() => {}, 100000);
```

### LINKED ISSUE #22724: `process.off` does not restore default SIGINT behaviour
Repro:

```
import process from "node:process";
console.log("ready");
setInterval(() => {}, 1000);
const listener = () => {
  console.log("foo");
  process.off("SIGINT", listener);
};
process.on("SIGINT", listener);
```

Node: correct

```
$ node /tmp/signal.js 
ready
^Cfoo
^C
$ 
```

Deno: wrong

```
$ deno run -A /tmp/signal.js 
ready
^Cfoo
^C^C^C^C^C^C^C^C
```

### PR DESCRIPTION
**Pull request by nathanwhit**
Fixes #22724. Fixes #7164.

This does add a dependency on `signal-hook`, but it's just a higher level API on top of `signal-hook-registry` (which we and `tokio` already depend on) and doesn't add any transitive deps.

### CODE REVIEW

**Comment by bartlomieju** [comment]
I believe this also fixes a long standing issue: https://github.com/denoland/deno/issues/7164

**Review by mmastrac — APPROVED** [review]
LGTM! Great fix.

File: runtime/ops/signal.rs [hunk 1]
```diff
@@ -606,6 +617,10 @@ pub fn op_signal_unbind(
   state: &mut OpState,
   #[smi] rid: ResourceId,
 ) -> Result<(), AnyError> {
-  state.resource_table.take_any(rid)?.close();
+  let resource = state.resource_table.take::<SignalStreamResource>(rid)?;
+  resource
```
Comment (mmastrac) [inline runtime/ops/signal.rs:621]: `#[cfg(unix)]`, likely

File: tests/unit_node/process_test.ts [hunk 2]
```diff
@@ -237,6 +237,7 @@ Deno.test({
 
 Deno.test({
   name: "process.off signal",
+  ignore: Deno.build.os == "windows",
```
Comment (mmastrac) [inline tests/unit_node/process_test.ts:240]: 👍 

LGTM, do we have an issue for process.off not working on Windows?

Comment (nathanwhit) [reply tests/unit_node/process_test.ts:240]: Dug a little deeper on this, turns out it's a `process.off` issue, it's a more general signal sending issue on windows (https://github.com/denoland/deno/issues/22846)

**Pull request merged**